scripts/ci/bounded_download.py

```
@@ -0,0 +1,45 @@
+    if max_bytes <= 0:
+        raise ValueError("download ceiling must be positive")
+    if destination.exists():
+        raise DownloadRejected("download destination already exists")
+    scratch = destination.with_name(f".{destination.name}.downloading")
+    if scratch.exists():
+        raise DownloadRejected("download scratch destination already exists")
+
+    request = urllib.request.Request(
+        url,
+        headers={"Accept": "application/octet-stream", "User-Agent": "assay-release-verifier"},
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)